core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -29,6 +30,16 @@ import org.apache.spark.serializer.{SerializerInstance, Serializer}
 import org.apache.spark.util.{CompletionIterator, Utils}
 
 /**
+ * Factory class that creates iterators to read records from fetched blocks
+ */
+trait BlockRecordIteratorFactory {
```

Comment:
Also, I think that any new shuffle interfaces should be implemented as Java interfaces, not Scala traits.